File: ChatTemplate/ClientApp/src/app/models/ship.ts
import { ShipType } from './ship-type';
import { Cell } from './cell';

export class Ship {
  
  public type: ShipType;
  public size: number;
  public hits: number;
  public coordinates: Cell[];
  public isDead: boolean;

  constructor(points: Cell[], type: ShipType = null) {
    this.type = type;
    this.coordinates = points;
    this.size = points.length;
    this.hits = 0;
    this.isDead = false;
  }
}

File: ChatTemplate/Models/Ship.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ChatTemplate.Models
{
    public class Ship
    {
        public ShipType type { get; set; }
        public int size { get; set; }
        public int hits { get; set; }
        public ICollection<Cell> coordinates { get; set; }
        public bool isDead { get; set; }
    }
}
